includes/Modules/Analytics_4.php

```diff
@@ -192,6 +207,21 @@ function ( $new_value, $old_value ) {
 			add_filter( 'googlesitekit_inline_modules_data', $this->get_method_proxy( 'inline_custom_dimensions_data' ) );
 		}
 
+		// Check if property ID has changed and reset the value for `propertyCreateTime`.
+		// @TODO Merge it with same filter above once `keyMetrics` feature flag is removed.
+		add_filter(
+			'pre_update_option_googlesitekit_analytics-4_settings',
+			function ( $new_value, $old_value ) {
+				if ( $new_value['propertyID'] !== $old_value['propertyID'] ) {
+					$new_value['propertyCreateTime'] = '';
+				}
+
+				return $new_value;
+			},
+			10,
+			2
+		);
```

Comment:
We probably don't need this filter since the setting will be set on the client when changing the connected property. It will cause a problem otherwise as mentioned in my previous comment.